packages/jaeger-ui/src/model/transform-trace-data.tsx

```diff
@@ -88,7 +88,18 @@ export default function transformTraceData(data: TraceData & { spans: SpanData[]
   const max = data.spans.length;
   for (let i = 0; i < max; i++) {
     const span: Span = data.spans[i] as Span;
-    const { startTime, duration, processID } = span;
+    const { startTime, duration, processID, tags } = span;
+
+    // avoid display value precision loss in case of value overflow Number.MAX_VALUE
+    if(tags) {
+      for(let j = 0; j < tags.length; j++) {
+        const tag = tags[j]
+        if(Number.isInteger(tag.value) && !Number.isSafeInteger(tag.value)) {
```

Comment:
What is the type of tag.value at this point? If it's a string, no need to convert, if it's a number then the precision may already be lost. I would've thought the fix needs to be on the server side such that big numeric numbers are sent as strings in JSON.

Reply:
>  "if it's number then the precision may already be lost" 

 but i got the right console output after i converted it to bigint, it confused me.
![image](https://user-images.githubusercontent.com/32982627/186059152-f3451d71-be17-4da2-a3a0-623c62c883c4.png)

api return 
![image](https://user-images.githubusercontent.com/32982627/186059653-fed1a2e2-4eba-4040-932e-09e55c819f03.png)

i'll consider output string when it is big numeric numbers. it's a way to solve it after all.

Reply:
- should server side return below data instead
```json
{
    "key": "xx_id",
    "type": "string",
    "value": "780177723283783680"
}
```

- or should jaeger json model return value as json string ?
https://github.com/jaegertracing/jaeger/blob/main/model/json/model.go#L100
```go
type KeyValue struct {
	Key   string      `json:"key"`
	Type  ValueType   `json:"type,omitempty"`
	Value interface{} `json:"value"` //json output string ?
}
```

all ways seems to be strange in some logic way 😂